src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let res = 0;
+
+  return function add(val) {
+    if (val === undefined) {
+      const totalRes = res;
+
+      res = 0;
+
+      return totalRes;
+    }
+
+    res += val;
+
+    return add;
+  };
 }
```

Comment:
Alternative solution
```suggestion
  const add = (value) => {
    if (value === undefined) {
      const totalRes = res;

      res = 0;

      return totalRes;
    }

    res += value;

    return add;
  }
  
  return add;
}
```